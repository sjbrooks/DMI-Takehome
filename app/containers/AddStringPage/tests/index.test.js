/**
 * Test the AddStringPage
 */

import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import { AddStringPage, mapDispatchToProps } from '../index';
import {
  changeString,
  createString,
  stringCreated,
  stringCreationError,
} from '../actions';
// import { loadRepos } from '../../App/actions';
import configureStore from '../../../configureStore';

describe('<AddStringPage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <AddStringPage
            loading={false}
            error={false}
            string=""
            created={false}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should submit string creation form if input form is not empty', async () => {
    const submitSpy = jest.fn();
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <AddStringPage
            string="Not Empty"
            onchangeString={() => {}}
            onSubmitForm={submitSpy}
          />
        </IntlProvider>
      </Provider>,
    );

    const input = container.querySelector('input');
    fireEvent.submit(input);
    expect(submitSpy).toHaveBeenCalled();
  });

  // NOTE: Although the onSubmit function will still be called in case of empty input,
  // the logic inside the saga prevents the creation of an empty string.
  // See saga.test.js for where we test an empty string.

  // it('should not create string if input form is submitted while empty', () => {
  //   const submitSpy = jest.fn();
  //   const { container } = render(
  //     <Provider store={store}>
  //       <IntlProvider locale="en">
  //         <AddStringPage
  //           string=""
  //           onchangeString={() => {}}
  //           onSubmitForm={submitSpy}
  //         />
  //       </IntlProvider>
  //     </Provider>,
  //   );

  //   const input = container.querySelector('input');
  //   fireEvent.submit(input);
  //   expect(submitSpy).not.toHaveBeenCalled();
  // });

  //   it('should not call onSubmitForm if string is an empty string', () => {
  //     const submitSpy = jest.fn();
  //     render(
  //       <Provider store={store}>
  //         <IntlProvider locale="en">
  //           <AddStringPage onchangeString={() => {}} onSubmitForm={submitSpy} />
  //         </IntlProvider>
  //       </Provider>,
  //     );
  //     expect(submitSpy).not.toHaveBeenCalled();
  //   });

  //   it('should not call onSubmitForm if string is null', () => {
  //     const submitSpy = jest.fn();
  //     render(
  //       <Provider store={store}>
  //         <IntlProvider locale="en">
  //           <AddStringPage
  //             string=""
  //             onchangeString={() => {}}
  //             onSubmitForm={submitSpy}
  //           />
  //         </IntlProvider>
  //       </Provider>,
  //     );
  //     expect(submitSpy).not.toHaveBeenCalled();
  //   });

  //   describe('mapDispatchToProps', () => {
  //     describe('onchangeString', () => {
  //       it('should be injected', () => {
  //         const dispatch = jest.fn();
  //         const result = mapDispatchToProps(dispatch);
  //         expect(result.onchangeString).toBeDefined();
  //       });

  //       it('should dispatch changeString when called', () => {
  //         const dispatch = jest.fn();
  //         const result = mapDispatchToProps(dispatch);
  //         const string = 'mxstbr';
  //         result.onchangeString({ target: { value: string } });
  //         expect(dispatch).toHaveBeenCalledWith(changeString(string));
  //       });
  //     });

  //     describe('onSubmitForm', () => {
  //       it('should be injected', () => {
  //         const dispatch = jest.fn();
  //         const result = mapDispatchToProps(dispatch);
  //         expect(result.onSubmitForm).toBeDefined();
  //       });

  //       it('should dispatch loadRepos when called', () => {
  //         const dispatch = jest.fn();
  //         const result = mapDispatchToProps(dispatch);
  //         result.onSubmitForm();
  //         expect(dispatch).toHaveBeenCalledWith(loadRepos());
  //       });

  //       it('should preventDefault if called with event', () => {
  //         const preventDefault = jest.fn();
  //         const result = mapDispatchToProps(() => {});
  //         const evt = { preventDefault };
  //         result.onSubmitForm(evt);
  //         expect(preventDefault).toHaveBeenCalledWith();
  //       });
  //     });
  //   });
});
