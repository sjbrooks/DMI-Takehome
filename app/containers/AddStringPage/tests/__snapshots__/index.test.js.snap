// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddStringPage /> should render and match the snapshot 1`] = `
<div>
  <h1
    class="H1-tncdg6-0 hpAKSC"
  >
    <span>
      Add a string
    </span>
  </h1>
  <form
    class="Form-oznt7c-0 kzXhZz"
  >
    <label
      for="username"
    >
      <span>
        Submit a word, phrase, or text blurb.
      </span>
      <input
        class="Input-sc-1pkdwb8-0 ljGaqD"
        id="string"
        placeholder="Enter text here"
        type="text"
        value=""
      />
    </label>
  </form>
</div>
`;
